### ISSUE TITLE
[Bug] `live_grep` displays the wrong (delayed) preview with fzf `0.40.0`

### ISSUE THREAD

**Issue by nyngwang** [ibhagwan/fzf-lua#769]
### Info

- Operating System: macOS 13.5
- Shell: zsh
- Terminal: Alacritty
- `nvim --version`: `NVIM v0.10.0-dev-387+gd31f7648e`
- `fzf --version`: `0.41.0(brew)`

- [x] The issue is reproducible with `mini.sh`

<details>
<summary>fzf-lua configuration</summary>

```lua
require('fzf-lua').setup({
})

vim.keymap.set('n', '<C-f>', function ()
  require('fzf-lua').live_grep_glob {
    cwd = vim.fn.getcwd(-1,0),
    actions = { ['ctrl-f'] = 'abort' }
  }
end)
```
</details>

### Description 

I got different results on grepping `<C-e`, and all of them are incorrect, i.e. it found `<C-w/v/x ..` instead.

https://github.com/ibhagwan/fzf-lua/assets/24765272/58cd0a8a-4c7b-4a1b-88a9-2f5dc4bf8750

**Comment by ibhagwan** [comment]
Could it be because `-` is a special regex character?

You need to escape it with `\-`.

**Comment by nyngwang** [comment]
The same result. The entry is correct but the output is wrong:

<img src="https://github.com/ibhagwan/fzf-lua/assets/24765272/8248812d-def9-4a6b-8463-d06ef77e54e3" width="500">

**Comment by ibhagwan** [comment]
The entry seems correct, it always ends on the same file `keymap.lua:177`, the issue here seems to be an **incorrect preview**, seems like this has something to do with the speed of typing as it shows the previous match preview.

**Comment by nyngwang** [comment]
> with the speed of typing as it shows the previous match preview.

But I typed it really slowly during testing, lol.

**Comment by ibhagwan** [comment]
> > with the speed of typing as it shows the previous match preview.
> 
> But I typed it really slowly during testing.

Maybe not the speed then but there is obviously some combo here causing the preview to not get updated from the previous result.

**Comment by nyngwang** [comment]
I suspect that some `autocmd`s of the floating window is not triggered, not sure though.

**Comment by ibhagwan** [comment]
> I suspect that some `autocmd`s of the floating window is not triggered, not sure though. I tried adding `vim.cmd('redraw')` in the code of previewer, didn't work.

I don’t think this is related, it seems to be some race condition of the fzf preview command trigger.

**Comment by ibhagwan** [comment]
> But I typed it really slowly during testing

looking at the video again you’re not typing slowly at all, you remove the `e` and then add the `e` back in an instant and the new preview doesn’t get updated, notice you’re still getting the preview from the search for `<C-` (without the `e`).

**Comment by nyngwang** [comment]
Yes, that demo is somehow misleading. But I just did another test after reading your comment regarding the typing speed. From my current trying, it's not related to typing speed.

**Comment by ibhagwan** [comment]
Ty @nyngwang, I figured it out, this is an upstream issue: https://github.com/junegunn/fzf/issues/3311

**Comment by ibhagwan** [comment]
FYI, this bug was apparently introduced with fzf `0.40.0`, if you downgrade to `0.39.0` this will be resolved.

**Comment by nyngwang** [comment]
Just confirmed that upgrading to `0.40.1`, according to the upstream issue you kindly created, fixes this problem :) Thanks for your help! This issue can be safely closed.

https://github.com/ibhagwan/fzf-lua/assets/24765272/8ad8d5d5-0e15-4706-9329-469e857256f1

**Comment by ibhagwan** [comment]
> Just confirmed that upgrading to `0.40.1`

Small correction the fixed version is `0.41.1`.

Ty @nyngwang, closing.
